### PR TITLE
Support Disabling Source Type Override Parameters, including Source Version

### PR DESCRIPTION
**Pull request by marcus-bcl**
*Issue #, if available:*
https://github.com/aws-actions/aws-codebuild-run-build/issues/57

*Description of changes:*
This PR is an extension of https://github.com/aws-actions/aws-codebuild-run-build/pull/64 to ensure all three source override parameters are disabled.  It also fixes merge conflicts and npm audit issues.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

File: code-build.js [hunk 1]
```diff
@@ -152,6 +152,8 @@ async function waitForBuildEndTime(
 
 function githubInputs() {
   const projectName = core.getInput("project-name", { required: true });
+  const disableSourceOverride =
+    core.getInput("disable-source-override") === "true";
```
Comment (taoyong-ty) [inline code-build.js:156]: For consistency, could you either update this new line to use `core.getInput("disable-source-override", {required: false}) === "true"`, or remove all other `{ required: false }`s from githubInputs()?

File: test/code-build-test.js [hunk 2]
```diff
@@ -104,6 +104,21 @@ describe("githubInputs", () => {
       .and.to.deep.equal(["one", "two", "three", "four"]);
   });
 
+  it("skips override when parameter is set to true", () => {
```
Comment (taoyong-ty) [inline test/code-build-test.js:107]: this test case name is a bit confusing. should we just called it `can process disable-source-override `

Comment (taoyong-ty) [inline test/code-build-test.js:107]: could you also add a assertion in [`build basic parameters for codeBuild.startBuild`](https://github.com/aws-actions/aws-codebuild-run-build/blob/master/test/code-build-test.js#L60) that expect(test).to.haveOwnProperty("disableSourceOverride").and.to.equal(false);

File: README.md [hunk 3]
```diff
@@ -31,6 +31,8 @@ The only required input is `project-name`.
 1. **image-override** (optional) :
    The name of an image for this build that overrides the one specified
    in the build project.
+1. **disable-source-override** (optional) :
+   Set to `true` if you want to disable providing `sourceTypeOverride` and `sourceLocationOverride` to CodeBuild.
```
Comment (taoyong-ty) [inline README.md:35]: and `sourceVersion`

**Review by taoyong-ty — COMMENTED** [review]
Thank you so much for making this enhancement. 

Apology for the delayed review. If you are still interested in wrapping up and merging this pull request, please address the comment and resolve the conflicts. 

I am ready to approve and release once the comments have been addressed

**Review by taoyong-ty — APPROVED** [review]

**Comment by taoyong-ty** [comment]
@marcus-bcl  could you resolve the conflict, so I can merge this PR ?

**Comment by marcus-bcl** [comment]
> @marcus-bcl could you resolve the conflict, so I can merge this PR ?

Apologies I didn't see that your approval, I've resolved the conflict now - feel free to merge it.  Thanks!

**Comment by fstehle** [comment]
Looks like this PR breaks existing usages.
Our CI fails now with the following error:
```
Run aws-actions/aws-codebuild-run-build@v1
*****STARTING CODEBUILD*****
Error: Unexpected key 'disableSourceOverride' found in params
*****CODEBUILD COMPLETE*****
```

**Comment by fstehle** [comment]
Ping @taoyong-ty

**Comment by tothmate** [comment]
> Looks like this PR breaks existing usages @taoyong-ty. Our CI fails now with the following error:
> 
> ```
> Run aws-actions/aws-codebuild-run-build@v1
> *****STARTING CODEBUILD*****
> Error: Unexpected key 'disableSourceOverride' found in params
> *****CODEBUILD COMPLETE*****
> ```

Same here

**Comment by DrewCurd** [comment]
Our CI too is failing with the above

**Pull request merged**